exercise1/terraform/main.tf

```diff
@@ -0,0 +1,505 @@
+  resource_group_name  = azurerm_resource_group.ex1.name
+  virtual_network_name = azurerm_virtual_network.ex1_vnet.name
+  address_prefixes     = ["10.0.0.0/24"]
+}
+
+resource "azurerm_network_security_group" "ex1_sql_netsecg" {
+  name                = "${var.rg_name}_pe_netsecg"
+  location            = azurerm_resource_group.ex1.location
+  resource_group_name = azurerm_resource_group.ex1.name
+}
+
+resource "azurerm_subnet_network_security_group_association" "ex1_secg_asso_pe" {
+  subnet_id                 = azurerm_subnet.ex1_subnet_pe.id
+  network_security_group_id = azurerm_network_security_group.ex1_sql_netsecg.id
+}
+
+resource "azurerm_private_dns_zone" "ex1_priv_dns_zone" {
+  name                = "${var.rg_name}.privatelink.database.windows.net"
+  resource_group_name = azurerm_resource_group.ex1.name
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "ex1_priv_dns_z_net_link" {
+  name                  = "${var.rg_name}_priv_dns_z_net_link"
+  resource_group_name   = azurerm_resource_group.ex1.name
+  registration_enabled  = true
+  private_dns_zone_name = azurerm_private_dns_zone.ex1_priv_dns_zone.name
+  virtual_network_id    = azurerm_virtual_network.ex1_vnet.id
+}
+
+resource "azurerm_private_endpoint" "ex1_redis_private_end" {
+  name                = "${var.rg_name}_redis_private_end"
+  resource_group_name = azurerm_resource_group.ex1.name
+  location            = azurerm_resource_group.ex1.location
+
+  subnet_id = azurerm_subnet.ex1_subnet_pe.id
+
+  private_service_connection {
+    name                           = "${var.rg_name}_redis_private_serv_conn"
+    private_connection_resource_id = azurerm_redis_cache.ex1_vm_redis.id
+    subresource_names              = ["redisCache"]
+    is_manual_connection           = false
+  }
+
+  private_dns_zone_group {
+    name                 = "${var.rg_name}_private_dns_zg_redis"
+    private_dns_zone_ids = [azurerm_private_dns_zone.ex1_priv_dns_zone.id]
+  }
+}
+
+resource "azurerm_private_endpoint" "ex1_sqldb_private_end" {
+  name                = "${var.rg_name}_sql_private_end"
+  resource_group_name = azurerm_resource_group.ex1.name
+  location            = azurerm_resource_group.ex1.location
+
+  subnet_id = azurerm_subnet.ex1_subnet_pe.id
+
+  private_service_connection {
+    name                           = "${var.rg_name}_sql_private_serv_conn"
+    private_connection_resource_id = azurerm_mssql_server.ex1_sql_server.id
+    subresource_names              = ["sqlServer"]
+    is_manual_connection           = false
+  }
+
+  private_dns_zone_group {
+    name                 = "${var.rg_name}_private_dns_zg_sql"
+    private_dns_zone_ids = [azurerm_private_dns_zone.ex1_priv_dns_zone.id]
+  }
+}
+
+# ********************** Database associated resources **************************************************
+
+#_______________________ SQL Database resources _______________________________________________
+resource "azurerm_storage_account" "ex1_store_acc" {
```

Comment:
I was looking into an implementation where the virtual machine extension calls the script from a file in the storage account then left this because i thought it was necessary for the `azurerm_mssql_database` which it seems that it is not. 

Removing